### ISSUE TITLE
Multiple similar filters can lose the existing filter

### ISSUE THREAD

**Issue by adevvv** [MorphiaOrg/morphia#1814]
I think there is a bug in the new Morphia 2.2.3. I translated code from Morphia 1.x and the generated MongoDB queries seem to be mixed up when combining .exits() and .not() and that kind of combinations.

```java
// OLD Morphia 1.x WORKED
q.or(q.criteria("disabled").doesNotExist(), q.criteria("disabled").hasThisOne(Boolean.FALSE));

// NEW Morphia 2.x DOES NOT WORK
q.filter(Filters.or(Filters.exists("disabled").not(), Filters.eq("disabled", Boolean.FALSE)));
);

// OLD Morphia 1.4 WORKED
//			q.or(
//					q.criteria("belongsToContentId").doesNotExist(),
//					q.and(
//							q.criteria("belongsToContentId").exists(),               
//							q.criteria("showAtGuideLevel").hasThisOne(Boolean.TRUE)
//					)
//			);

// NEW Morphia 2.x DOES NOT WORK

	q.filter(
			Filters.or(
					Filters.exists("belongsToContentId").not(),
					Filters.and(
						Filters.exists("belongsToContentId"),           
						Filters.eq("showAtGuideLevel", Boolean.TRUE)
					)
			));
```

**Comment by adevvv** [comment]
Please let me elaborate.
1) I am building a Filter with 
`Query<Event> q = dbs.getDatastore().createQuery(Event.class);`

2) then I add some filter params
```
// OLD Morphia 1.4 q.or(q.criteria("status").doesNotExist(), q.criteria("status").equal(Constants.EVENT_STATUS_UPCOMING));
		q.filter(
				Filters.or(
						Filters.exists("status").not(),
						Filters.eq("status",Constants.EVENT_STATUS_UPCOMING)
				)
		);
```
Query evaluates to:
Q1: MorphiaQuery[clazz=Event, query=Document{{customerId=577d2f8ee4b0646495dd09cb, $or=[Document{{status=Document{{$exists=false}}}}, Document{{status=0}}]}}]

3) I add more filter params

```
// OLD Morphia 1.4
//			q.or(
//					q.criteria("belongsToContentId").doesNotExist(),
//					q.and(
//							q.criteria("belongsToContentId").exists(),               
//							q.criteria("showAtGuideLevel").hasThisOne(Boolean.TRUE)
//					)
//			);

			q.filter(
					Filters.or(
							Filters.exists("belongsToContentId").not(),
							Filters.and(
								Filters.exists("belongsToContentId"),              
								Filters.eq("showAtGuideLevel", Boolean.TRUE)
							)
					));
		}
```

Query evaluates to: 

Q2: MorphiaQuery[clazz=Event, query=Document{{customerId=577d2f8ee4b0646495dd09cb, $or=[Document{{belongsToContentId=Document{{$exists=false}}}}, Document{{$and=[Document{{belongsToContentId=Document{{$exists=true}}}}, Document{{showAtGuideLevel=true}}]}}]}}]

Why did the query lose the "Document{{status=0}}" from Step 2) to 3)?
Using Morphia core 2.2.3

        <dependency>
            <groupId>dev.morphia.morphia</groupId>
            <artifactId>morphia-core</artifactId>
            <version>2.2.3</version>
        </dependency>

**Comment by adevvv** [comment]
One more question: Datastore.createQuery is marked as @Deprecated since 2.0. If this is not the intended way to build up complex queries that can add x filters on the way (not at once!), how can one do it with Morphia then?

**Comment by evanchooly** [comment]
You should use `find()` instead:  https://morphia.dev/morphia/2.2/javadoc/dev/morphia/Datastore.html#createQuery(java.lang.Class)

**Comment by adevvv** [comment]
Thanks for the hint, will use in the future:
`Query<Event> q = dbs.getDatastore().find(Event.class);`
instead of 
`Query<Event> q = dbs.getDatastore().createQuery(Event.class);`

Would you be so kind as to share the priority for this bug or in what release it will be handled?

**Comment by evanchooly** [comment]
I haven't had a chance to look at it (i'm caught up in a few other things) but I'd imagine this will go in a 2.2.4 soonish.

But while i have you, what do you think of the `blah().not()` syntax vs `not(blah())` ?  The former is the outcome of some earlier thinking (and this API was always intended to be malleable until the `experimental` is taken off) and, based on the evolution of some of the other APIs, I'm thinking the latter probably feels a bit more natural especially when you have larger expressions you're trying to `not`.  Have any strong feelings on that one?

**Comment by adevvv** [comment]
Thanks for clarifying and that you intent to fix this soon, I sense if the q-"pipe" gets destroyed or looses state, that's a horrific error for a Query-Lib. 

Concerning the not, I would stick to syntax of MongoDB or SQL, but cannot now tell you what they do. Thanks for a great project!

**Comment by evanchooly** [comment]
At this point, I think the general shape is fixed.  No one's complained about the syntax so i'm taking that as at least tacit approval.  When that experimental tag comes off, it'll be a package naming change.  I haven't determined how feasible it will be to have both in place for a transition period but I'm hoping I can pull that off.  I want it to be as painless as possible.

**Comment by adevvv** [comment]
I tried today if the problem is fixed with morphia 2.2.5, but there still seems to be a problem.
STEP 1: 
```
final Query<Event> q = dbs.getDatastore().find(Event.class);
q.filter("customerId", customerId);
		q.filter(
				Filters.or(
						Filters.exists("status").not(),
						Filters.eq("status",0
				)
		);
log.info("Q1: " + q);
```
>> Q1: MorphiaQuery[clazz=Event, query=Document{{customerId=5ca0c24a4ac35d7748610e68, $or=[Document{{status=Document{{$exists=false}}}}, Document{{status=0}}]}}]

STEP 2: 
```
			q.filter(
					Filters.or(
							Filters.exists("belongsToContentId").not(),
							Filters.and(
								Filters.exists("belongsToContentId"),              
								Filters.eq("showAtGuideLevel", Boolean.TRUE)
							)
					));
log.info("Q2: " + q);
```
>>> Q2: MorphiaQuery[clazz=Event, query=Document{{customerId=5ca0c24a4ac35d7748610e68, $or=[Document{{belongsToContentId=Document{{$exists=false}}}}, Document{{$and=[Document{{belongsToContentId=Document{{$exists=true}}}}, Document{{showAtGuideLevel=true}}]}}]}}]

**_BUG: The filter of status from before is gone?_**
